lib/api/core/entrypoints/embedded/protocols/socketio.js

```diff
@@ -123,16 +123,15 @@ class SocketIoProtocol extends Protocol {
   onClientDisconnection(clientId) {
     debug('[%s] onClientDisconnection', clientId);
 
-    if (this.sockets[clientId]) {
-      delete this.sockets[clientId];
-      delete this.entryPoint.clients[clientId];
-
-      return this.entryPoint.removeConnection(clientId);
+    if (this.sockets.has(clientId)) {
+      this.sockets.delete(clientId);
+      this.entryPoint.removeConnection(clientId);
```

Comment:
The actual fix is here: we don't delete `this.entryPoint.clients` entries before invoking `removeConnection` anymore